### ISSUE TITLE
extremely high variance in transformer_big loss 

### ISSUE THREAD

**Issue by ARozental** [tensorflow/tensor2tensor#889]
### Description
I'm creating a language model over Twitter with the transformer_big parameter set. The only change I made is max_length=64 (because tweets are usually short). 
I'm using 8 Tesla V100 GPUs and my loss looks like that after 0.99 smoothing on Tensor Board:
 
![screen shot 2018-06-25 at 15 08 14](https://user-images.githubusercontent.com/9210444/41849488-cf6a1222-7889-11e8-9660-6f923a00a9c0.png)

From what I understand the effective batch size is 4096*8 and the loss I see is averaged over ~32K examples. Because of that I would think that the difference in the loss between two following batches will be extremely small, however even after over 250K batches I often see losses as high as I have gotten at step 10K.

I tried shuffling the data manually, increasing batch size to 5K and reducing learning rate to 0.01 but all of it doesn't seem to help.
Have you any idea as to what can cause it?

**Comment by martinpopel** [comment]
> the loss I see is averaged over ~32K examples

No. For variable-length problems the batch_size is specified in tokens (subwords), not sentences. So your batch with at most 4096*8 tokens (some of which are padding) corresponds to about 1k sentences if the average sentence size is 32 tokens.
Note also that TensorBoard shows an average over multiple batches (usually 100, I think) and in case of loading an event file with many datapoints it loads just a subsample, which affects the apparent variance.

I would say that high variance in the training loss is not bad *per se*, thought it is good to be aware of it. You should watch the dev-set learning curve of your target intrinsic or even better extrinsic evaluation measure.

That said, I've never done language modelling on Twitter, so I cannot judge whether your training-loss curve is reasonable/expected.

**Comment by ARozental** [comment]
Thank you for making the meaning of batch_size clearer me. The model does learn well (enough to generate reasonable tweets) and I do average over 100 batches in this Tensor Board screenshot. 
Still while averaging over ~1000*100 tweets I would not expect ~20% change in the loss of subsequent units of 100 batches.
